packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -710,8 +710,8 @@ async function updateCloudFormationNestedStack(
   const backEndDir = pathManager.getBackendDirPath();
   const projectRoot = pathManager.findProjectRoot();
   const rootStackFilePath = path.join(pathManager.getRootStackBuildDirPath(projectRoot), rootStackFileName);
-  // deploy new nested stack to disk
-  JSONUtilities.writeJson(rootStackFilePath, nestedStack);
+  // deploy preprocess nested stack to disk
+  storeRootStackTemplate(context, nestedStack);
```

Comment:
does this function use writeJson utilities? If yes, then await?

Reply:
nice catch!